redisinsight/api/src/modules/rdi/client/api.rdi.client.ts

```diff
@@ -37,6 +39,29 @@ import { RdiResetPipelineFailedException } from '../exceptions/rdi-reset-pipelin
 import { RdiStartPipelineFailedException } from '../exceptions/rdi-start-pipeline-failed.exception';
 import { RdiStopPipelineFailedException } from '../exceptions/rdi-stop-pipeline-failed.exception';
 
+// The structure of `config` varies based on the RDI schema definition,
+// but it always includes a `sources` key.
+// Example:
+// {
+//   sources: {
+//     some_database_source: {
+//       host: 'localhost',
+//       port: 6379,
+//       ...other connection options
+//     }
+//   }
+// }
+interface ConnectionsConfig {
+  sources: Record<string, Record<string, unknown>>;
+}
+
+const prepareTestSourcesConnectionsOptions = (config: ConnectionsConfig) => {
+  const sourcesOptions = config.sources || {};
```

Comment:
does it make sense to add `?` here `config?.sources || {};` if we assume that we might have some unexpected config

Reply:
I tried few cases, but we always get a `config`. You can send a `config`, without a `sources`, but we got this scenario covered. Anyways this is inside try catch, so we're safe.

<img width="655" alt="Screenshot 2025-02-14 at 16 00 02" src="https://github.com/user-attachments/assets/79df656b-d805-4eb5-9f6c-71cae2cccc56" />